MINOR: AbstractConfigTest.testClassConfigs should reset the class loa…

chia7712 opened the pull request:
reference to https://github.com/apache/kafka/pull/10006#issuecomment-770394615

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

chia7712 committed:
MINOR: AbstractConfigTest.testClassConfigs should reset the class loader to avoid impacting mock/spied object

showuon commented on clients/src/test/java/org/apache/kafka/common/config/AbstractConfigTest.java:
```diff
@@ -276,52 +277,46 @@ public RestrictedClassLoader() {
+        ClassLoader origin = Thread.currentThread().getContextClassLoader();
         try {
-            new ClassTestConfig(ClassTestConfig.RESTRICTED_CLASS.getName(), null);
-            fail("Config created with class property that cannot be loaded");
```
It's good refactor to use `assertThrow`, but the failed message is missed. Please add it in `assertThrow`. Thanks.

showuon commented on clients/src/test/java/org/apache/kafka/common/config/AbstractConfigTest.java:
```diff
@@ -276,52 +277,46 @@ public RestrictedClassLoader() {
-        try {
-            testConfig = new ClassTestConfig(null, Arrays.asList(ClassTestConfig.VISIBLE_CLASS.getName(), ClassTestConfig.RESTRICTED_CLASS.getName()));
-            testConfig.getConfiguredInstances("list.prop", MetricsReporter.class);
-            fail("Should have failed to load class");
```
same here

showuon commented on clients/src/test/java/org/apache/kafka/common/config/AbstractConfigTest.java:
```diff
@@ -276,52 +277,46 @@ public RestrictedClassLoader() {
-        try {
-            testConfig = new ClassTestConfig(null, ClassTestConfig.VISIBLE_CLASS.getName() + "," + ClassTestConfig.RESTRICTED_CLASS.getName());
-            testConfig.getConfiguredInstances("list.prop", MetricsReporter.class);
-            fail("Should have failed to load class");
```
same here

showuon reviewed (commented):
@chia7712 , Thanks for the fix. Left some minor comments.

chia7712 committed:
Merge branch 'trunk' into MINOR-10010

chia7712 committed:
add message to assertThrow

chia7712 commented:
@showuon Thanks for your reviews. I have addressed your comment by https://github.com/apache/kafka/pull/10010/commits/dac031f228c8e8db02de2f17e14f43a96db11b8e

showuon reviewed (approved):
LGTM

hachikuji commented on clients/src/test/java/org/apache/kafka/common/config/AbstractConfigTest.java:
```diff
@@ -276,52 +277,49 @@ public RestrictedClassLoader() {
-        testConfig = new ClassTestConfig(ClassTestConfig.RESTRICTED_CLASS, Arrays.asList(ClassTestConfig.VISIBLE_CLASS, ClassTestConfig.RESTRICTED_CLASS));
-        testConfig.checkInstances(ClassTestConfig.RESTRICTED_CLASS, ClassTestConfig.VISIBLE_CLASS, ClassTestConfig.RESTRICTED_CLASS);
-        // Properties specified as classNames should fail to load classes
+        ClassLoader origin = Thread.currentThread().getContextClassLoader();
```
nit: maybe `originalClassLoader` to be consistent with other names?

hachikuji reviewed (approved):
LGTM. Left a minor naming suggestion.

chia7712 committed:
Merge branch 'trunk' into MINOR-10010

chia7712 committed:
consistent naming

chia7712 committed:
trivial change to trigger QA
